src/Models/NonhydrostaticModels/NonhydrostaticModels.jl

```diff
@@ -6,10 +6,20 @@ using KernelAbstractions.Extras.LoopInfo: @unroll
 using Oceananigans.Utils: launch!
 using Oceananigans.Grids
 using Oceananigans.Solvers
+using Oceananigans.Distributed: MultiArch, DistributedFFTBasedPoissonSolver, reconstruct_global_grid   
 using Oceananigans.ImmersedBoundaries: ImmersedBoundaryGrid
 
 import Oceananigans: fields, prognostic_fields
 
+function PressureSolver(arch::MultiArch, grid::RegRectilinearGrid)
+    full_grid = reconstruct_global_grid(grid)
+    if arch.ranks[1] == 1 && full_grid.Nx == full_grid.Ny  # we would have to allow different settings 
+        return DistributedFFTBasedPoissonSolver(arch, full_grid, grid)
```

Comment:
```suggestion
function PressureSolver(arch::MultiArch, local_grid::RegRectilinearGrid)
    global_grid = reconstruct_global_grid(local_grid)
    if arch.ranks[1] == 1 && global_grid.Nx == global_grid.Ny  # we would have to allow different settings 
        return DistributedFFTBasedPoissonSolver(arch, global_grid, local_grid)
```

Reply:
Can take of that `global_grid.Nx == global_grid.Ny` actually because of #2079